src/Resolvers/Microsoft.NET.Sdk.WorkloadManifestReader/WorkloadManifestReader.Localization.cs

```diff
@@ -0,0 +1,101 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Globalization;
+using System.Runtime.CompilerServices;
+using Microsoft.NET.Sdk.Localization;
+
+
+#if USE_SYSTEM_TEXT_JSON
+using System.Text.Json;
+#else
+using JsonTokenType = Newtonsoft.Json.JsonToken;
+#endif
+
+namespace Microsoft.NET.Sdk.WorkloadManifestReader
+{
+    partial class WorkloadManifestReader
+    {
+        private static LocalizationCatalog ReadLocalizationCatalog(ref Utf8JsonStreamReader reader)
+            => new LocalizationCatalog(ReadStringDictionary(ref reader));
+
+        public static string? GetLocalizationCatalogFilePath(string manifestFilePath, CultureInfo? culture = null)
+        {
+            culture ??= CultureInfo.CurrentUICulture;
```

Comment:
according to https://github.com/dotnet/sdk/blob/a30e465a2e2ea4e2550f319a2dc088daaafe5649/src/Cli/dotnet/UILanguageOverride.cs#L27

we should read DefaultThreadCurrentUICulture

Reply:
A thread will use that if it does not have an explicit culture set: https://github.com/dotnet/runtime/blob/cd1b4cff818a700b6ff82fde024c65b036979894/src/libraries/System.Private.CoreLib/src/System/Globalization/CultureInfo.cs#L394